app/assets/src/components/AlignmentViz.jsx

```diff
@@ -0,0 +1,276 @@
+import axios from "axios";
+import React from "react";
+
+class ReadViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.parsedAlignment = this.parseAlignment(props);
+  }
+
+  parseAlignment(readInfo) {
```

Comment:
I'd recommend the following two steps:

1. If `parseAlignment` only depends on its input argument `readInfo` and not any other instance props (`this.props`), we can extract it to a separate file and import the function into this class to use. 

This way it's easier to write test. And also, since the function doesn't depend on class instances and their specific properties, keeping the function outside of the class will keep the class cleaner.

So we can create a file `parse-alignment.js` and move this function into it. Like so:

```js
function parseAlignment(readInfo) {
  ...
};

export default parseAlignment;
```

And then in the current class file, we can just import the function, like so:

```jsx
import parseAlignment from './parse-alignment'
```

2. Now that we have the function as a separate file, we can create a test file in the same directory like so: `parse-alignment.test.js`, and then code:

```js
import jest from 'jest'
import parseAlignment from './parse-alignment'

describe('parseAlignment', () => {
  it('renders the expected snapshot', () => {
    const readInfo = ... // Not sure what the input shape looks like
    expect(parseAlignment(readInfo)).toMatchSnapshot()
  })
})
```
3. Now you've created a test for the function, we need to run test to create a snapshot of the output object. So run `npm test` and Jest will generate a snapshot file for you!

4. Next time you need to modify the function, you can run test and see if the output is still expected :-)

Some syntax might be off, because I'm at a conference now and don't have access to my laptop XD Hope this helps!

Reply:
Yes! We just need to modify the code above slightly by instantiating an instance of the `ReadViz` class. Like so:

```tsx
import jest from 'jest'
import ReadViz from './ReadViz'

describe('ReadViz', () => {
  describe('parsedAlignment', () => {
    it('matches the expected snapshot', () => {
      const readInfo = ... // Not sure what the input shape looks like
      const instance = new ReadViz(readInfo)

      expect(instance.parsedAlignment).toMatchSnapshot()
    })
  })
})
```

This should work. If not, please let me know 👌